components/style/media_queries.rs

```diff
@@ -243,19 +243,13 @@ pub fn parse_media_query_list(input: &mut Parser) -> MediaList {
     }
 
     let mut media_queries = vec![];
-    let mut found_invalid = false;
     loop {
         match input.parse_until_before(Delimiter::Comma, MediaQuery::parse) {
-            Ok(mq) => if !found_invalid {
+            Ok(mq) => {
                 media_queries.push(mq);
             },
-            Err(..) => if !found_invalid {
-                media_queries.clear();
+            Err(..) => {
                 media_queries.push(MediaQuery::never_matching());
```

Comment:
Actually, don't we need to fail parsing on certain kinds of errors (like `@media 2px { ... }`) but not others?  It looks like these changes will accept any unexpected input and still consider the media rule to have been parsed correctly.